src/components/TransactionOperation.tsx

```diff
@@ -1,60 +1,156 @@
-      <p>
-        {!t.isIncoming && (
-          <span className="icon has-text-success">
-            <i className="fas fa-arrow-down" />
-          </span>
-        )}
-        {t.isIncoming && (
-          <span className="icon has-text-danger">
-            <i className="fas fa-arrow-up" />
+const getAmount = (t: TransactionOperationView) => {
+  let amount = 0
+  switch (t.operation.type) {
+    case 'create_account':
+      amount = Number(t.operation.starting_balance)
+      break
+    case 'payment':
+      amount = Number(t.operation.amount)
+      break
+    default:
+      return 'Other'
+  }
+  return t.isIncoming ? amount : amount + Number(t.fee)
+}
+
+interface StateProps {
+  moreInfoIsHidden: boolean
+  toggleMoreInfo: () => any
+}
+const TransactionCard: React.SFC<Props & StateProps> = ({
+  transactionWithOperation: t,
+  currency,
+  moreInfoIsHidden,
+  toggleMoreInfo,
+}) => (
+  <article className="level">
+    <div
+      className="columns box level-item is-gapless has-text-grey-lighter is-multiline"
+      style={{ width: '100%' }}
+    >
+      <div className="column is-4">
+        <div className="is-flex" style={{ alignItems: 'center' }}>
+          {isTransfer(t.operation) ? (
+            t.isIncoming ? (
+              <span className="icon is-large has-text-success">
+                <i className="fal fa-lg fa-arrow-down" />
+              </span>
+            ) : (
+              <span className="icon is-large has-text-danger">
+                <i className="fal fa-lg fa-arrow-up" />
+              </span>
+            )
+          ) : (
+            <span className="icon is-large has-text-grey-light">
+              <i className="fal fa-lg fa-dash" />
+            </span>
+          )}
+          <span className="has-text-weight-bold">
+            <AddressView address={getAddress(t)} />
           </span>
-        )}
-        <span>{startCase(t.operation.type)}</span>
-      </p>
-    </div>
-    <div className="message-body">
-      {renderOperationRecords(t.operation)}
-      <HorizontalLabelledField label="Fee" value={t.fee} isCompact={true} />
-      <HorizontalLabelledField label="Memo" value={t.memo} isCompact={true} />
+        </div>
+      </div>
+      <div className="column is-3">
+        <p>{t.memo || '(no description)'}</p>
+      </div>
+      <div className="column is-2">
+        <span className="icon is-small has-text-success">
+          <i className="fal fa-xs fa-check-circle" />
+        </span>
+        <span>{t.date.toTimeString().slice(0, 8)}</span>
+      </div>
+      <div className="column is-2">
+        <p
+          className={`${
+            t.isIncoming ? 'has-text-success' : 'has-text-danger'
+          } has-text-weight-bold has-text-right`}
+        >
+          {getAmount(t)} {currency}
+        </p>
+      </div>
+      <div className="column is-1 has-text-right">
+        <button className="button is-text" onClick={toggleMoreInfo}>
+          {moreInfoIsHidden && (
```

Comment:
this could just update the `<i />` className attribute based on `moreInfoIsHidden` instead of also the wrapping span etc? 

eg. 
```typescript
    <span className="icon">
      <i className={`fal fa-lg fa-angle-${moreInfoIsHidden ? 'down' : 'up'}`} />
    </span>
```

Reply:
Unfortunately font-awesome doesn't like that, since it directly edits the HTML instead of the virtual DOM of react. Doesn't actually update.